Wikipedia/Code/WMFAppViewController.m

```diff
@@ -1211,15 +1211,19 @@ - (BOOL)processUserActivity:(NSUserActivity *)activity animated:(BOOL)animated c
             [self dismissPresentedViewControllers];
             [self setSelectedIndex:WMFAppTabTypeMain];
             [self.navigationController popToRootViewControllerAnimated:NO];
-            WMFPreferredLanguagesViewController *languagesVC = [WMFPreferredLanguagesViewController
-                    preferredLanguagesViewController];
+            WMFPreferredLanguagesViewController *languagesVC = [WMFPreferredLanguagesViewController preferredLanguagesViewController];
             languagesVC.showExploreFeedCustomizationSettings = YES;
             languagesVC.delegate = self.settingsViewController;
             [languagesVC applyTheme:self.theme];
             [self showSettingsWithSubViewController:languagesVC animated:animated];
         } break;
         default: {
             NSURL *linkURL = [activity wmf_linkURL];
+            MWKLanguageLink *preferredLanguage = self.dataStore.languageLinkController.appLanguage;
```

Comment:
It would be better to use -preferredLanguageVariantCodeForLanguageCode: here.

You can call:
`linkURL.wmf_languageVariantCode = [linkController.preferredLanguageVariantCodeForLanguageCode: linkURL.wmf_language]` and it will do the correct thing if `wmf_language` is nil, if the language code doesn't have a variant, etc.

Reply:
Also, it will handle any language that comes in, even though we're pretty sure it will be in the user's primary language.

Reply:
What good is having new API if we don't even use it? 😉 Thanks for flagging this – updated.